desmod/simulation.py

```diff
@@ -205,7 +220,7 @@ def simulate(config, top_type, env_type=SimEnvironment, reraise=True,
 
 
 def simulate_factors(base_config, factors, top_type,
-                     env_type=SimEnvironment, jobs=None):
+                     env_type=SimEnvironment, jobs=None, only_factor=None):
```

Comment:
What if instead of having this new `only_factor` argument, we instead had a filter function? I.e. the user would pass in a function that evaluated whether each config was either included or not. E.g.:

    simulate_factors(..., config_filter=lambda config: config['meta.sim.index'] == 42)

This would make this feature more general. I could, for example, envision this being useful for at least two other applications:

- Selecting more than one index: `config_filter=lambda c: c['meta.sim.index'] in [42, 142]`.
- Sampling configs from a very large configuration space: `config_filter: lambda c: random.random() < 0.1`. I.e. pick a random 10% of the factorial configs.

I think it would be more reasonable to push this more general capability into `factorial_config()` than the specific `only_factor` capability. Unfortunately, I don't think it would afford the same performance optimization that plumbing `only_factor` affords.

N.B. if we pushed this filter function capability down to `factorial_config()`, that function would have to take responsibility for assigning `meta.sim.index`. We currently do that assignment in `simulate_factors()`.